### PR TITLE
[PLACES-338] Patch LDAP injection vulnerability on login

### PR DESCRIPTION
**Pull request by aiyan**
Currently, the Express authentication middleware runs an LDAP filter using the username provided in the login request. The username should be checked for special characters used in search filters so that an attacker cannot perform an injection attack.

These changes use a library to escape all special characters in the username.

@sherifkozman Let me know what information you would need to perform the migration.

### CODE REVIEW

**Comment by sherifkozman** [comment]
@MarshallMcCoy @aiyan i updated the tests with the correct values , tests still fail though.

https://github.com/Path-Check/safeplaces-backend/runs/778557357?check_suite_focus=true

**Comment by aiyan** [comment]
Very strange. Tests passed on my computer. @sherifkozman could it be the environment variables? I changed both the LDAP server and the middleware.

image

File: .github/workflows/intergation-tests.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ env:
   LDAP_ORG: ${{ secrets.LDAP_ORG }}
   LDAP_BIND: 'cn=admin,dc=covidsafepaths,dc=org'
   LDAP_SEARCH: 'cn={{username}}, dc=covidsafepaths, dc=org'
```
Comment (aiyan) [inline .github/workflows/intergation-tests.yaml:44]:
```suggestion
  LDAP_SEARCH: 'dc=covidsafepaths, dc=org'
```

**Comment by sherifkozman** [comment]
@aiyan how are you passing the credentials during the tests?

**Comment by aiyan** [comment]
@sherifkozman I pass the credentials in the same way as before, just as an `application/json` with fields `username` and `password`.

**Comment by aiyan** [comment]
@sherifkozman Any clue why the tests are timing out? I think it may be the LDAP server - are the tests using the updated LDAP server? I made changes to it

**Comment by MarshallMcCoy** [comment]
@aiyan we saw a similar issue with another one of our PRs yesterday. Jeff found that changing the `describe` blocks to not use arrow functions turned out to be the solution to fixing the timeouts.

Example:

```
change describe('honors expires at on previously published case', () => {

to

this describe('honors expires at on previously published case', function() {
```

**Comment by aiyan** [comment]
@MarshallMcCoy Wow, that is so strange. But even stranger is that `login.test.js` does not contain arrow functions. 😭

**Comment by MarshallMcCoy** [comment]
@aiyan you could try and extend the timeout on the tests?

**Comment by aiyan** [comment]
I talked with @sherifkozman, and he said that the CI tests are against OpenLDAP, not the mock LDAP JS server. Maybe we can examine the config employed by OpenLDAP

**Comment by MarshallMcCoy** [comment]
@aiyan maybe rebase off of `master`? I think should at least allow the app to build in the `Build/Publish` step.

**Comment by aiyan** [comment]
_Finally the tests pass!_ 🍰 🍰 🍰
@MarshallMcCoy

**Pull request merged**